Login, Logout and quota disco rework

RakuJa opened the pull request.

RakuJa committed:
Bozza login e logout

RakuJa committed:
Api -> Network model

RakuJa committed:
Login/Logout

RakuJa committed:
Clean up maincontroller

RakuJa committed:
Partly fixes #56,Completely fixes #36,Completely fixes #47

RakuJa committed:
Closes #36

RakuJa committed:
Add type hint

RakuJa committed:
Add settings model to os handler

RakuJa committed:
Fix test

yukiLu6 committed:
TEST

RakuJa committed:
Revert "TEST"

This reverts commit c8561fa5c6195884a541b9eb8090aa051da4d33b.

RakuJa committed:
Revert "Revert "TEST""

This reverts commit 54ffed3ec28e97504c97fa025bd05a5f05b579bd.

yukiLu6 committed:
Aggiunta gestione download in caso di poco spazio rimasto nella cartella condivisa

segnatevelo nei calendari, non farò mai più un commit così pieno di cose, e non l'ho nemmeno fatto da sola, quindi non so quanto valga

yukiLu6 committed:
fix del commit di prima

eh, volevate che facessi tutto giusto

RakuJa committed:
Fix test

RakuJa committed:
Add login button test

RakuJa committed:
Cambio unità di misura ( Kb => KiB)

RakuJa committed:
Rework set quota disco

RakuJa committed:
Completely closes #57, partly fixes #54

RakuJa committed:
Fix test

RakuJa committed:
fixes #61, partly fixes #54

Matteo- committed:
cambiata procedura login

Edivad99 commented on src/algorithm/decision_engine.py:
```diff
@@ -1,26 +1,31 @@
-from src.model.algorithm.policy import Policy
-from src.network.api_exceptions import APIException
-from src.network.api import Api
+from ..controllers.notification_controller import NotificationController
```
mettere `src.controllers` in linea con tutto il resto del codice

Edivad99 commented on src/controllers/notification_controller.py:
```diff
@@ -5,14 +5,15 @@
         self.notification_view.show()
 
-    def send_message(self, msg: str, duration: int = 1000) -> None:
+        self.send_message("Bentornato %s" % username)
```
No pls😂

Edivad99 commented on src/network/api_implementation.py:
```diff
@@ -72,89 +71,45 @@ def check_status_code(self, response):
-        return ""
-
     @ExceptionsHandler
     def is_logged(self, _cookie: str = "") -> bool:
```
Perchè in questo metodo non settiamo `self.cookie`? Non settandolo la prossima call di rete che fa va a richiedere un nuovo cookie

Edivad99 commented:
Domani cerco di fare una gestione intelligente sull'invio delle notifiche, anche perchè se un utente cambia il path della cartella o esegue la prima sync ha uno spam di notifiche assurdo

RakuJa committed:
fixes #60

RakuJa committed:
fix #63

RakuJa committed:
Fix network test

RakuJa committed:
partly fixes #58

Edivad99 committed:
Merge branch 'develop' into Login-and-logout-rework

Edivad99 committed:
Update requirements.txt

Edivad99 committed:
Fix di molti errori nell chiamate di funzioni

Login_controller, effettua il login senza cliccare sul bottone
Commentata notifica di benvenuto (troppo spammosa)
Eliminato login with cookie, con la nuova api conviene fare una chiamata e ricevere il cookie più aggiornato rispetto a tentare il login e vedere se il cookie è ancora valido
Per sicurezza login_screen non inserisce la password nel campo di testo
